### PR TITLE
[extension/awsproxy] add support for shutdown test

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Change the lifecycle of the extension so that the creation of the AWS client happens in the Start method, which helps with supporting component lifecycle.

**Link to tracking Issue:**
Relates to #27849

### CODE REVIEW

File: extension/awsproxy/extension.go [hunk 1]
```diff
@@ -25,6 +25,12 @@ type xrayProxy struct {
 var _ extension.Extension = (*xrayProxy)(nil)
 
 func (x xrayProxy) Start(_ context.Context, _ component.Host) error {
+	srv, err := proxy.NewServer(&x.config.ProxyConfig, x.settings.Logger)
```
Comment (crobert-1) [inline extension/awsproxy/extension.go:28]: Thanks for moving this from `create` to `Start`, really helpful 🎉

File: .chloggen/awsproxy_lifecycle.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: awsproxyextension
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Make awsproxy extension support shutdown
```
Comment (crobert-1) [inline .chloggen/awsproxy_lifecycle.yaml:10]: Out of ignorance, I don't understand how this adds support for shutdown. Shutdown itself already existed, what's the user-facing change that's worth documenting here? 

I assume the lack of a `nil` check before is what blocked the generated component test from being enabled?

Comment (atoulme) [reply .chloggen/awsproxy_lifecycle.yaml:10]: It would fail at creation time. The message is not great, sorry.

Comment (atoulme) [reply .chloggen/awsproxy_lifecycle.yaml:10]:
```suggestion
note: Make awsproxy extension more resilient by initiating the AWS client during Start.
```

**Review by crobert-1 — APPROVED** [review]

**Comment by atoulme** [comment]
@Aneurysm9, @mxiamxia, would you please review as codeowners? Thanks!

**Comment by atoulme** [comment]
Adding for discussion to SIG meeting 4/3.

File: extension/awsproxy/extension.go [hunk 3]
```diff
@@ -25,6 +25,12 @@ type xrayProxy struct {
...
+
+	if err != nil {
+		return err
+	}
+	x.server = srv
```
[the remark this answers is not in the document]
Comment (atoulme) [reply extension/awsproxy/extension.go:33]: Thanks! I have updated the PR to use pointer receivers on Start and Shutdown.

**Review by Aneurysm9 — APPROVED** [review]
I don't think this actually tests anything, but it doesn't seem to hurt anything, either. v0v

**Pull request merged**